src/api/app/controllers/webui/badge_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Webui::BadgeController < Webui::WebuiController
+  before_action :set_project
+  before_action :require_package
+
+  def show
+    results = @package.buildresult(@project, false, true).results[@package.name]
+    render inline: create_badge(results)
+  end
+
+  private
+
+  def create_badge(results)
+    filename = 'badge-unknown.svg'
+    return Rails.application.assets[filename].source if results.empty?
```

Comment:
```suggestion
    return Rails.application.assets[filename].source if results.blank?
```
Where there aren't any build results, `results` is `nil`.

Reply:
Without any build results, leaving the code with `empty?`, it errors with:
```
Completed 500 Internal Server Error in 167ms (ActiveRecord: 9.7ms | Backend: 7.5ms | Allocations: 110903)
NoMethodError (undefined method `empty?' for nil:NilClass

    return Rails.application.assets[filename].source if results.empty?
                                                               ^^^^^^^):

app/controllers/webui/badge_controller.rb:14:in `create_badge'

```